yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -1153,13 +1178,81 @@ func (p *ParserIssueDetector) detectTablePKRecommendations(tm *TableMetadata) []
 		}
 	}
 
-	if len(options) > 0 {
-		issues = append(issues, NewMissingPrimaryKeyWhenUniqueNotNullIssue("TABLE", tm.GetObjectName(), options))
+	// Process UNIQUE constraints
+	for _, uniqueCols := range tm.GetUniqueConstraints() {
+		processUniqueColumns(uniqueCols)
+	}
+
+	// Process UNIQUE indexes (column-based only)
+	for _, uniqueCols := range tm.GetUniqueColumnBasedIndexes() {
+		processUniqueColumns(uniqueCols)
+	}
+
+	// Remove duplicates from options using lo.UniqBy
+	uniqueOptions := lo.UniqBy(options, func(option []string) string {
+		return strings.Join(option, ",")
+	})
+
+	if len(uniqueOptions) > 0 {
+		issues = append(issues, NewMissingPrimaryKeyWhenUniqueNotNullIssue("TABLE", tm.GetObjectName(), uniqueOptions))
 	}
 
 	return issues
 }
 
+// getAllPartitionColumnsInHierarchy returns all partition columns down the entire hierarchy
+// for a root-level partitioned table. This includes partition columns from all child tables.
+func (p *ParserIssueDetector) getAllPartitionColumnsInHierarchy(tm *TableMetadata) []string {
```

Comment:
why do you need this? I don't see any usage of this 
I think you already have this with `TableMetadata`